src/mumble/ServerHandler.cpp

```diff
@@ -745,7 +745,7 @@ void ServerHandler::serverConnectionConnected() {
 	qscCipher = connection->sessionCipher();
 
 	if (!qscCert.isEmpty()) {
-		const QSslCertificate &qsc = qscCert.last();
+		const QSslCertificate &qsc = qscCert.first();
```

Comment:
In this step we essentially obtain the certificate the server used for encrypting its TLS session, right? And if we used `last` instead, we would instead end up with the client's root CA?

If so, we could add a comment like this:
```suggestion
		// Get the server's immediate SSL certificate
		const QSslCertificate &qsc = qscCert.first();
```

Reply:
Yes we get the certificate the server uses for TLS here, but since we are dealing with the server's certificate chain here `last` would not give the client's root CA, but the server's.

Quick note for the sake of completeness: According to the TLS standard, peers are allowed to omitt root certificates, as long as they provide all intermediate certificates leading to the root certificate. I guess the idea here is that the other peer must already know that root certificate anyway to accept the chain.